test/create_exercise_test.dart

```diff
@@ -128,5 +129,349 @@ void main() {
+    - no_adjacent_strings_in_list
+    - valid_regexps
+'''));
+      });
+
+      group('testCaseTemplate tests', () {
+        test('simple test for a single case', () {
+          Map<String, dynamic> testCase = <String, dynamic>{
+            "description": "Zero is an Armstrong number",
+            "property": "isArmstrongNumber",
+            "input": {"number": 0},
+            "expected": true
+          };
+
+          expect(testCaseTemplate('armstrong-numbers', testCase), equals('''
+    test('${testCase['description']}', () {
+      final null result = armstrongNumbers.${testCase['property']}(${testCase['input']['number']});
+      expect(result, equals(${testCase['expected']}));
+    }, skip: false);
+'''));
+        });
+      });
+    });
+
+    group('processing test cases', () {
+      test('single case', () {
+        List<Map<String, dynamic>> cases = <Map<String, dynamic>>[
+          <String, dynamic>{
+            "description": "Say Hi!",
+            "property": "hello",
+            "input": <String, dynamic>{},
+            "expected": "Hello, World!"
+          }
+        ];
+
+        Set<dynamic> actualSet = retrieveListOfExpected(cases);
+        Set<String> expectedSet = new Set.from(<String>['Hello, World!']);
+
+        expect(actualSet, equals(expectedSet));
+      });
+
+      test('multiple cases', () {
+        List<Map<String, dynamic>> cases = <Map<String, dynamic>>[
+          <String, dynamic>{
+            "description": "year not divisible by 4 in common year",
+            "property": "leapYear",
+            "input": {"year": 2015},
+            "expected": false
+          },
+          <String, dynamic>{
+            "description": "year divisible by 2, not divisible by 4 in common year",
+            "property": "leapYear",
+            "input": {"year": 1970},
+            "expected": false
+          },
+          <String, dynamic>{
+            "description": "year divisible by 4, not divisible by 100 in leap year",
+            "property": "leapYear",
+            "input": {"year": 1996},
+            "expected": true
+          },
+          <String, dynamic>{
+            "description": "year divisible by 100, not divisible by 400 in common year",
+            "property": "leapYear",
+            "input": {"year": 2100},
+            "expected": false
+          },
+          <String, dynamic>{
+            "description": "year divisible by 400 in leap year",
+            "property": "leapYear",
+            "input": {"year": 2000},
+            "expected": true
+          },
+          <String, dynamic>{
+            "description": "year divisible by 200, not divisible by 400 in common year",
+            "property": "leapYear",
+            "input": {"year": 1800},
+            "expected": false
+          }
+        ];
+
+        Set<dynamic> actualSet = retrieveListOfExpected(cases);
+        Set<bool> expectedSet = new Set.from(<bool>[false, true]);
```

Comment:
Recommend: Use `Set.of([false, true])`, this allows to remove the type annotations altogether.

Example `final expectedSet = Set.of([false, true]);`